### PR TITLE
feat: Fit implements collections.abc.Mapping

### LINKED ISSUE #142: Iterating over an instance of `Fit` raises AttributeError.
Instances of `stan.fit.Fit` have a `__getitem__` method and a `__len__` method. But iterating over them will raise a weird AttributeError: 
```
AttributeError: 'int' object has no attribute 'endswith'
```

Given how `len`  and `__getitem__` work,  I think __iter__ should scan through the names of the parameters.

Perhaps there is a good reason for not making a fit object be iterable, in which case I think `__iter__` should raise a `TypeError` instead:

```
TypeError: 'stan.fit.Fit' object is not iterable
```

### PR DESCRIPTION
**Pull request by riddell-stan**
The Fit class now implements collections.abc.Mapping. Users can iterate
over a Fit instance and use several of the normal dictionary methods.

Closes #142

### CODE REVIEW

File: stan/fit.py [hunk 1]
```diff
@@ -139,16 +134,30 @@ def __getitem__(self, param):
...
+
+    def items(self) -> Generator[Tuple[str, np.ndarray], None, None]:
+        for name in self.keys():
+            yield name, self[name]
+
+    def __iter__(self) -> Generator[str, None, None]:
+        return self.keys()
+
     def __len__(self) -> int:
         return len(self.param_names)
```
Comment (ahartikainen) [inline stan/fit.py:162]: do we want to use

```suggestion
        return len(self.keys())
```

**Review by ahartikainen — APPROVED** [review]
One comment, but otherwise, looks good.

File: stan/fit.py (hunk 1 above)
Comment (riddell-stan) [reply stan/fit.py:162]: keys() is a generator so it has no length. But you're right, we should properly implement the dictionary spec and return DictionaryViews (or whatever it is that dictionaries use).

Comment (riddell-stan) [reply stan/fit.py:152]: Nevermind. I did this wrong initially. It's fixed now.

You get keys, values, items methods merely by inheriting from collections.abc.Mapping and implementing _getitem__, __iter__, __len__.

**Review by ahartikainen — APPROVED** [review]

**Pull request merged**